### PR TITLE
New commands and improvements

### PR DESCRIPTION
**Pull request by fredericojordan**
- Adds the following commands:
  - `groups`
  - `group_create`
  - `groups_clear`
  - `devices_clear`
- Adds `--bondid` option to the following commands (ignores selected bond):
  - `devices`
  - `device_create`
  - `token`
  - `upgrade`
  - `version`
- Adds `-q` option for removing table outline from output for the following commands:
  - `devices`
  - `discover`
  - `list`
- Adds the `--force` option to the `upgrade` command, removing the need for user input
- Ignores all entries in devices/groups list starting with `_` (fixes `__` bug)
- Increases table column width from `16` to `20`
- Fixes `flake8` long line warning
- Runs [`Black`](https://github.com/psf/black) linter on code base (with some exceptions where `flake8` broke)
- Bumps version from `0.0.14` to `0.1.0` (see [semver.org](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase))

### CODE REVIEW

File: bond/cli/table.py [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
 
-    def print_boarder(self):
-        print("-" + "-" * ((self.col_width + 1) * len(self.header) - 1) + "-")
+    def print_border(self):
+        if not self.quiet:
+            print("-" * ((self.col_width + 1) * len(self.header) + 1))
 
     def print_header(self):
-        self.print_boarder()
+        self.print_border()
```
Comment (chrismerck) [inline bond/cli/table.py:29]: :-)

**Review by chrismerck — DISMISSED** [review]
Thanks, Fred. Good to see this scale of improvements / new capabilities coming into the "cli".

**Comment by chrismerck** [comment]
One other comment:  at this point we just need it to work for our QA purpose, but it would be good to have a scheme/plan for how the command naming scales to the various things we will do. `device_delete` vs `devices_clear` maybe `device_delete_all` is clearer? idk. As long as it is consistent somehow.

I will say that I was hoping we would have a second level of menus, something like:

```
bond device create
bond device delete
```

without the underscore.

and

```
bond device
```

presumably would show the various device-related subcommands.

like `git`, which everyone knows is the perfectly intuitive CLI ;)

**Comment by fredericojordan** [comment]
> One other comment: at this point we just need it to work for our QA purpose, but it would be good to have a scheme/plan for how the command naming scales to the various things we will do. `device_delete` vs `devices_clear` maybe `device_delete_all` is clearer? idk. As long as it is consistent somehow.
> 
> I will say that I was hoping we would have a second level of menus, something like:
> 
> ```
> bond device create
> bond device delete
> ```
> 
> without the underscore.
> 
> and
> 
> ```
> bond device
> ```
> 
> presumably would show the various device-related subcommands.
> 
> like `git`, which everyone knows is the perfectly intuitive CLI ;)

@chrismerck 
Yep, makes sense to refactor some of those commands.

Right now the command `devices_clear` allows for the option `--deviceid` for a single device deletion, but I guess it could be improved.

Maybe:
```shell
bond device delete <device_id>
bond device delete --all
```
is a better scheme, but then `<device_id>` would be required only in the absence of the `--all` option, and I'm not sure I we can make the requirement conditional on other options, I'll have to look into it.

**Comment by chrismerck** [comment]
> bond device delete <device_id>
> bond device delete --all

that makes a lot of sense to me

File: bond/commands/groups_clear.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            "help": "force deletion with no input from user",
+            "action": "store_true",
+        },
+    }
+
+    def run(self, args):
+        bond_id = args.bondid or BondDatabase.get_assert_selected_bondid()
+
+        group_ids = bond.proto.get(bond_id, topic="groups").get("b", {})
+        group_ids.pop("_", None)
```
Comment (endy-s) [inline bond/commands/groups_clear.py:22]: Is there a way to "popRegex" every entry that starts with `_`?  To include `__` and `_modified`

File: bond/commands/groups_clear.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import bond.proto
+from bond.commands.base_command import BaseCommand
+from bond.database import BondDatabase
+
+
+class GroupsClearCommand(BaseCommand):
+    subcmd = "groups_clear"
+    help = "Remove all Bond's devices."
```
Comment (endy-s) [inline bond/commands/groups_clear.py]: Shouldn't it be group?
```suggestion
    help = "Remove all Bond's groups."
```

**Review by endy-s — DISMISSED** [review]

File: bond/commands/groups_clear.py (hunk 2 above)
Comment (fredericojordan) [reply bond/commands/groups_clear.py:22]: Yep, we discussed this after this PR was submitted so I'm including it in the next one! 😉

File: bond/commands/groups_clear.py (hunk 3 above)
Comment (fredericojordan) [reply bond/commands/groups_clear.py]: It should! Thanks!

File: bond/commands/groups_clear.py (hunk 2 above)
Comment (fredericojordan) [reply bond/commands/groups_clear.py:22]: Added here: #53

**Review by marciogranzotto — APPROVED** [review]
~It would be great to have `--bondID` on the `upgrade` and `livelog` commands too!~ 
Nevermind! I see you did that on another PR

**Review by endy-s — APPROVED** [review]

**Pull request merged**